autolens/pipeline/phase/abstract/result.py

```diff
@@ -31,6 +33,62 @@ def __init__(
     def most_likely_tracer(self):
         return self.analysis.tracer_for_instance(instance=self.instance)
 
+    @property
+    def source_plane_light_profile_centres(self):
+        if self.most_likely_tracer.source_plane.has_light_profile:
+            return self.most_likely_tracer.light_profile_centres_of_planes[-1]
+        else:
+            return []
+
+    @property
+    def source_plane_inversion_centres(self):
+        if self.most_likely_tracer.source_plane.has_pixelization:
+            return [
+                self.most_likely_fit.inversion.brightest_reconstruction_pixel_centre
+            ]
+        else:
+            return []
+
+    @property
+    def source_plane_centres(self):
+        centres = (
+            self.source_plane_light_profile_centres
+            + self.source_plane_inversion_centres
+        )
+        return grids.Coordinates(coordinates=[centres])
+
+    @property
+    def image_plane_multiple_image_positions_of_source_plane_centres(self):
+
+        # TODO : In the future, the multiple image positions functioon wil use an in-built adaptive grid.
+
+        grid = self.analysis.masked_dataset.mask.geometry.unmasked_grid
+
+        return list(
+            map(
+                lambda centre: self.most_likely_tracer.image_plane_multiple_image_positions(
+                    grid=grid, source_plane_coordinate=centre
+                ),
+                self.source_plane_centres[0],
```

Comment:
Always looks a bit dodgy to be just taking the first item from some array.

Reply:
The problem is what the tracer method returns. Positional mapping is still something we're figuring out how to actually do so this will be addresed once we've settled on the algorithm for that.